src/mjolnir/pbfgraphparser.cc

```diff
@@ -188,6 +188,11 @@ struct graph_callback : public OSMPBF::Callback {
       }
     };
 
+    tag_handlers_["layer"] = [this]() {
+      auto z_level = static_cast<int8_t>(std::stoi(tag_.second));
```

Comment:
do we have a try catch around all of the handlers when they get called? i didnt go check, but if we don't then we need to worry about that here. when someone puts `layer=chicken` we need to ignore it

Reply:
It seems we don't https://github.com/valhalla/valhalla/blob/c9ddd4bfa631b7dc44939bc35bde8017159712c2/src/mjolnir/pbfgraphparser.cc#L1229

But I see many other places where we use `std::stoi` in tag handlers. Should we add try/catch [here](https://github.com/valhalla/valhalla/blob/c9ddd4bfa631b7dc44939bc35bde8017159712c2/src/mjolnir/pbfgraphparser.cc#L1229)?

Reply:
i swore that we did a bunch of work to avoid unhandled exceptions when doing exactly this. now im wondering where the hell that code is!? is it possible that this stuff gets thrown out in lua if its not a number? to me it makes sense to put a try/catch there where yoou are suggesting. @gknisely @merkispavel where are we handling garbage tag values these days in the parser?

Reply:
Added try-catch https://github.com/valhalla/valhalla/pull/3261/files#diff-5fcdff37eb551f814e2a44f24f9c6bcc37856bb8dadbbcb59e93d173e46c1132R1234